boltstub/__init__.py

```diff
@@ -163,7 +163,10 @@ def start(self):
             self.server.server_close()
 
     def _close_socket(self):
-        self.server.socket.close()
+        if self.script.context.restarting or self.script.context.concurrent:
+            self.server.shutdown()
+        else:
```

Comment:
This change makes the existence of `_stop_server` superfluous.

I'd be surprised if this works as self.server.shutdown waits for all threads (`join`). So `try_skip_to_end` would block right at the beginning before being able to tell all the actors to try and skip to the end.